rotkehlchen/chain/bitcoin/bch/manager.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+from rotkehlchen.chain.bitcoin.bch.constants import HASKOIN_BASE_URL
+from rotkehlchen.chain.bitcoin.bch.utils import cash_to_legacy_address
+from rotkehlchen.chain.bitcoin.manager import BitcoinCommonManager
+from rotkehlchen.chain.bitcoin.types import BtcApiCallback, BtcQueryAction
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import BTCAddress, SupportedBlockchain
+from rotkehlchen.utils.misc import satoshis_to_btc
+from rotkehlchen.utils.network import request_get, request_get_dict
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BitcoinCashManager(BitcoinCommonManager):
+
+    def __init__(self, database: 'DBHandler') -> None:
+        haskoin_api = BtcApiCallback(name='haskoin', function_name='_query_haskoin_{0}')
+        # TODO: add other APIs for fallbacks if haskoin goes down
+        super().__init__(
+            database=database,
+            blockchain=SupportedBlockchain.BITCOIN_CASH,
+            api_callbacks={
+                BtcQueryAction.BALANCES: [haskoin_api],
+                BtcQueryAction.HAS_TRANSACTIONS: [haskoin_api],
+                # TODO: add support for querying transactions
+            },
+        )
+
+    @staticmethod
+    def _get_haskoin_address(
+            address_from_api: str,
+            requested_addresses: Sequence[BTCAddress],
+    ) -> BTCAddress | None:
+        """Get an address from a haskoin API response.
+        This is needed because the API only returns CashAddr format.
+        """
+        address: BTCAddress | None
+        if (
+            (address := BTCAddress(address_from_api)) in requested_addresses or
+            (address := BTCAddress(address_from_api.split(':')[1])) in requested_addresses or
+            (address := cash_to_legacy_address(address_from_api)) is not None
+        ):
+            return address
+
+        return None
+
+    def _query_haskoin_balances(
+            self,
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, FVal]:
+        """Query a mapping of addresses to balances from haskoin."""
```

Comment:
add the errors that this can raise. At least KeyError, deserializationError and RemoteError for what I can see. Also if those aren't handled they should be